notebooks/ReFT_Adapters_Finetuning.ipynb

```diff
@@ -0,0 +1,505 @@
+     ]
+    }
+   ],
+   "source": [
+    "from adapters import AutoAdapterModel, LoReftConfig\n",
+    "model = AutoAdapterModel.from_pretrained(model_name_or_path)\n",
+    "\n",
+    "config = LoReftConfig(r = 1, prefix_positions = 1, dropout = 0.05)\n",
+    "model.add_adapter(\"loreft_adapter\", config=config)\n",
+    "model.add_classification_head(\"loreft_adapter\", num_labels=3)\n",
+    "model.train_adapter(\"loreft_adapter\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "================================================================================\n",
+      "Name                     Architecture         #Param      %Param  Active   Train\n",
+      "--------------------------------------------------------------------------------\n",
+      "loreft_adapter           reft                 36,888       0.030       1       1\n",
+      "--------------------------------------------------------------------------------\n",
+      "Full model                               124,645,632     100.000               0\n",
+      "================================================================================\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(model.adapter_summary())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Evaluation\n",
+    "\n",
+    "We'll use accuracy as our main metric to evaluate the perforce of the reft model on the `mnli` dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import evaluate\n",
+    "import numpy as np\n",
+    "accuracy_metric = evaluate.load(\"accuracy\")\n",
+    "\n",
+    "def compute_metrics(eval_pred):\n",
+    "    logits, labels = eval_pred\n",
+    "    predictions = np.argmax(logits, axis=-1)\n",
+    "    return accuracy_metric.compute(predictions=predictions, references=labels)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Training\n",
+    "\n",
+    "Now we are ready to train our model. We will use the same set of hyper-parameters used to train the `mnli` dataset in the original paper except for the number of epochs to train on."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from adapters import AdapterTrainer\n",
+    "from transformers import TrainingArguments, Trainer\n",
+    "\n",
+    "training_args = TrainingArguments(\n",
+    "    output_dir='./results',\n",
+    "    eval_strategy='epoch',\n",
+    "    learning_rate=6e-4,\n",
+    "    per_device_train_batch_size=32,\n",
+    "    per_device_eval_batch_size=32,\n",
+    "    num_train_epochs=2,\n",
```

Comment:
please make a note that usually you'd train longer and this is only for demo purposes.